### DISCUSSION TITLE
choco upgrade all incredibly slow

### DISCUSSION THREAD

**Discussion by aagha** [chocolatey/choco#3180] [General] [answered]
My `choco upgrade all` command is incredibly slow!  I recall a while back I'd run it and it'd get through 40 programs in 30-40 seconds--now, it's running for 3-5 minutes!

What might this be a function of, and how could I diagnose what's happening?

**Comment by pauby** [comment, marked as answer]
Please see the [upgrade guide section on performance](https://docs.chocolatey.org/en-us/guides/upgrading-to-chocolatey-v2-v6#optimizing-performance).
